lib/reporter.js

```diff
@@ -37,416 +41,506 @@ const errorHandler = err => {
-        }
-        const currentStepIndex = testObj.steps.findIndex(step => step.name === testAssertion.assertion);
-        const currentStep = testObj.steps[currentStepIndex];
-        testObj.steps.splice(currentStepIndex, 1);
-
-        if (!currentStep) {
-            return;
-        }
-        const actualError = error || testAssertion.error;
-        if (actualError) {
-            // Logs error message for the failed steps
-            this.logMessage(currentStep.stepId, actualError.message, 'ERROR');
-        }
-
-        const additionalData = {};
-        if (testAssertion.skipped) {
-            additionalData.status = TestStatus.SKIPPED;
-            if (this.options.skippedIssue === false) {
-                additionalData.issue = { issueType: 'NOT_ISSUE' };
-            }
-        }
-
-        this.client
-            .finishTestItem(currentStep.stepId, {
-                status: currentStep.status || (actualError ? TestStatus.FAILED : TestStatus.PASSED),
-                ...additionalData
-            })
-            .promise.catch(errorHandler);
-        this.collectionMap.set(testAssertion.cursor.ref, testObj);
+    const parentId = this.getCurrentSuiteTempId();
+    const description = result.request.description && result.request.description.content;
+    const codeRefTitle = `${this.collectionRunOptions.collection.name}/${result.item.name}`;
+    const codeRef = utils.getCodeRef(this.collectionPath, codeRefTitle);
+    const parameters = utils.getParameters(
+      this.collectionRunOptions.iterationData,
+      result.cursor.iteration,
+    );
+
+    const testObj = this.client.startTestItem(
+      {
+        name,
+        type: 'TEST',
+        description,
+        codeRef,
+        parameters,
+        testCaseId: utils.getCollectionVariablesByKey(
+          'testCaseId',
+          this.collectionRunOptions.environment.values,
+        ),
+        attributes: utils.getAttributes(this.collectionRunOptions.environment.values),
+      },
+      this.launchObj.tempId,
+      parentId,
+    );
+
+    testObj.promise.catch(errorHandler);
+
+    this.sendRequestLogs(testObj.tempId, result.request);
+    this.collectionMap.set(result.cursor.ref, {
+      testId: testObj.tempId,
+      requestId: result.cursor.httpRequestId || result.item.id,
+      steps: [],
+      status: utils.getCollectionVariablesByKey(
+        'status',
+        this.collectionRunOptions.environment.values,
+      ),
+    });
+
+    this.testLogs &&
+      this.testLogs.forEach((log) =>
+        this.logMessage(testObj.tempId, log.message, log.level, log.time),
+      );
+    this.testLogs = [];
+
+    this.launchLogs &&
+      this.launchLogs.forEach((log) => this.sendLaunchLogMessage(log.message, log.level, log.time));
+    this.launchLogs = null;
+
+    this.suiteLogs &&
+      this.suiteLogs.forEach((log) => this.logMessage(parentId, log.message, log.level, log.time));
+    this.suiteLogs = null;
+  }
+
+  // Starts test scripts as test steps
```

Comment:
Please update comment as we don't start steps in this method anymore.
And I think it's event better to rename method to f.e. `collectStepsAdditionalData` or smth like that.
It's relevant for other event handlers as well.